### PR TITLE
Replace OpenTracing instrumentation with OpenTelemetry in grpc storage plugin 

### PR DESCRIPTION
**Pull request by afzal442**
## Which problem is this PR solving?
* Part of #3381 
* This PR adds `otelgrpc` plugin to storage

## Short description of the changes
- Replaces `otgrpc` plugin storage with otel grpc interceptors

### CODE REVIEW

**Review by albertteoh — APPROVED** [review]

File: plugin/storage/grpc/README.md [hunk 1]
```diff
@@ -200,13 +200,13 @@ grpc.ServeWithGRPCServer(&shared.PluginServices{
...
-        googleGRPC.UnaryInterceptor(otgrpc.OpenTracingServerInterceptor(tracer)),
-        googleGRPC.StreamInterceptor(otgrpc.OpenTracingStreamServerInterceptor(tracer)),
+        grpc.UnaryInterceptor(otelgrpc.UnaryServerInterceptor(otelgrpc.WithTracerProvider(tracerProvider))),
+		grpc.StreamInterceptor(otelgrpc.StreamServerInterceptor(otelgrpc.WithTracerProvider(tracerProvider))),
     })
 })
 ```
 
-Refer to `example/memstore-plugin` for more details.
+Refer to `example/hotrod` for more details.
```
[the remark this answers is not in the document]
Comment (afzalbin64) [reply plugin/storage/grpc/README.md:209]: hotrod contains the otel grpc implementation. we can allow users to refer to it instead.

[the remark this answers is not in the document]
Comment (yurishkuro) [reply plugin/storage/grpc/README.md:209]: and what is the main topic of this README?

File: plugin/storage/grpc/config/config.go [hunk 2]
```diff
@@ -84,8 +84,10 @@ func (c *Configuration) Close() error {
 
 func (c *Configuration) buildRemote(logger *zap.Logger) (*ClientPluginServices, error) {
 	opts := []grpc.DialOption{
-		grpc.WithUnaryInterceptor(otgrpc.OpenTracingClientInterceptor(opentracing.GlobalTracer())),
-		grpc.WithStreamInterceptor(otgrpc.OpenTracingStreamClientInterceptor(opentracing.GlobalTracer())),
+		grpc.WithUnaryInterceptor(
+			otelgrpc.UnaryClientInterceptor(otelgrpc.WithTracerProvider(otel.GetTracerProvider()))),
```
[the remark this answers is not in the document]
Comment (afzalbin64) [reply plugin/storage/grpc/config/config.go:88]: sorry Yuri! which other dependency you are talking abt? jtracer?

[the remark this answers is not in the document]
Comment (yurishkuro) [reply plugin/storage/grpc/config/config.go:88]: you're adding instrumentation that requires tracer provider (the "dependency"). It should be passed down from main, not pulled from a global

[the remark this answers is not in the document]
Comment (afzalbin64) [reply plugin/storage/grpc/config/config.go:88]: inside _factory.go_

File: plugin/storage/grpc/factory_test.go [hunk 3]
```diff
@@ -43,7 +44,7 @@ type mockPluginBuilder struct {
 	err        error
 }
 
-func (b *mockPluginBuilder) Build(logger *zap.Logger) (*grpcConfig.ClientPluginServices, error) {
+func (b *mockPluginBuilder) Build(logger *zap.Logger, tracer trace.TracerProvider) (*grpcConfig.ClientPluginServices, error) {
```
Comment (afzalbin64) [inline plugin/storage/grpc/factory_test.go:47]: I suspect we have to mock this tracer.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**